src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -311,6 +311,8 @@ object ICollection.SyncRoot
 
         // Virtual function for TreeSubSet, which may need to update its count.
         internal virtual void VersionCheck() { }
+        internal virtual void VersionCheckCount() { }
+        internal virtual int TotalCount() { return Count; }
```

Comment:
Why is this TotalCount wrapper around Count needed?

Reply:
https://github.com/acerbusace/corefx/blob/6564eb78f7fbe1408c8dbd82a498bb572e017c0f/src/System.Collections/src/System/Collections/Generic/SortedSet.cs#L1955

`TotalCount` is being used to get the parent set's **Count** property (when enumerating with a `TreeSubSet`). This way, enumeration does not require the Count property of TreeSubSet, hence no extra time wasted to updating the Count property first.

If there is another more efficient way, I can make the necessary changes.

Reply:
Ah I see. I got confused and thought you just introduced the parent's TotalCount, missed the overriden version.